actions/changelog-generate/action.yaml

```
@@ -0,0 +1,137 @@
+name: "Generate changelog"
+description: "Generate changelog after release"
+inputs:
+  # required inputs
+  github-token:
+    description: "The GitHub token for committing the changes."
+    required: true
+  new-tag:
+    description: Version after bump
+    required: true
+
+  # unrequired inputs
+  changelog-file:
+    description: Path to the Changelog.md file in the github repository
+    required: false
+    default: "CHANGELOG.md"
+  commitMode:
+    description: "Special configuration for projects which work without PRs."
+    required: false
+    default: "false"
+  config:
+    description: Path to the changelog config JSON file
+    required: false
+    default: "./.github/changelog-config.json"
+  fetchReleaseInformation:
+    description: "Will enable fetching additional release information from tags."
+    required: false
+    default: "false"
+  fetchReviewers:
+    description: "Will enable fetching the users/reviewers who approved the PR"
+    required: false
+    default: "false"
+  old-tag:
+    description: Previous version. Let empty for releases
+    required: false
+    default: ""
+  output:
+    description: "Name of the output file for the generate changelog step"
+    required: false
+    default: "changes.md"
+outputs:
+  merged-changelog:
+    description: "Changelog containing listing of all single changelogs"
+    value: ${{ steps.output-generator.outputs.merged-changelog }}
+  single-changelog:
+    description: "Changelog containing changes of the latest tag"
+    value: ${{ steps.output-generator.outputs.single-changelog }}
+runs:
+  using: "composite"
+  steps:
+    - name: "Set tags"
+      id: setTags
+      run: |
+        tag_list=$(git describe --tags) || tag_list=""
+
+        # to-tag
+        if [[ "$tag_list" == *"${{ inputs.new-tag }}"* ]]; then
+          echo  "totag=${{ inputs.new-tag }}">> $GITHUB_OUTPUT
+        else
+          latest_commit=`git rev-parse HEAD`
+          echo  "totag=$latest_commit">> $GITHUB_OUTPUT 
```

No need to introduce a variable if you only use it once:
```suggestion
          echo  "totag=$(git rev-parse HEAD)">> $GITHUB_OUTPUT 
```